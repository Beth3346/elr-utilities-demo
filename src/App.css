.main-content {
  max-width: 1200px;
  margin: auto;
  padding: 20px 0;
}

.header-inner {
  max-width: 1200px;
  margin: auto;
}

.sidebar-menu {
  padding: 20px 0;
  list-style: none;
}

.sidebar-menu li {
  padding: 5px 0;
}
