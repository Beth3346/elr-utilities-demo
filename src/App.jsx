import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import NavigationMain from '@/components/NavigationMain/NavigationMain';
import Home from '@/views/Home/Home';
import CSS from '@/views/CSS/CSS';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavigationMain />
        <Route exact path="/" component={Home} />
        <Route exact path="/css" component={CSS} />
      </div>
    );
  }
}

export default App;
