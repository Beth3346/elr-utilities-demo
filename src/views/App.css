.main-content {
  max-width: 1200px;
  margin: auto;
  padding: 20px 0;
}

.header-inner {
  max-width: 1200px;
  margin: auto;
}

.sidebar-menu {
  padding: 20px 10px 20px 0;
  list-style: none;
}

.sidebar-menu li {
  padding: 5px 0;
}

.sidebar-menu > li > ul {
  list-style: none;
  padding-left: 5px;
  font-size: 14px;
}

.main-content {
  display: flex;
  justify-content: flex-start;
}

.content-holder {
  padding: 30px 60px 30px 90px;
  max-width: 960px;
}

.content-holder p {
  padding: 5px 0;
  line-height: 1.5;
}

.example-heading {
  font-size: 36px;
  margin-bottom: 24px;
  padding-top: 0;
}

.example-heading,
.example-subheading {
  border-bottom: 1px solid #ccc;
  padding-bottom: 10px;
  font-weight: 700;
}

.example-subheading {
  padding: 10px 0;
  margin: 10px 0;
  font-size: 24px;
  color: #1976d2;
}
