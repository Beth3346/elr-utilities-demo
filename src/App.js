import React, { Component } from 'react';
import Header from './Header';
import NavigationMain from './NavigationMain';
import SidebarMain from './SidebarMain';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavigationMain />
        <Header heading="ELR Utilities">
          A collection of reusable, modular Sass Mixins
        </Header>
        <main className="main-content">
          <SidebarMain />
          <article className="content-holder">
            <section className="content">
              <h2 className="example-heading">Get Started</h2>
              <h3 className="example-subheading">
                Why Create a Sass Mixin Library
              </h3>
              <p>
                ELR Utilities is a framework of Sass Mixins and CSS designed to
                speed up and streamline the development process.
              </p>
              <p>
                I created this utility library because I found myself spending a
                lot of time looking up CSS syntax for stuff I used all the time.
                Especially clearfix. Using mixins also helps with consistency in
                CSS styles. If I decide I want to use a different method for
                styling or learn some new techniques I can update 1 mixin
                instead of scouring my entire CSS file.
              </p>
              <h3 className="example-subheading">
                The Future of ELR Utilities and CSS Libraries
              </h3>
              <p>
                While I do still think that Sass and CSS Libraries are useful
                for producivity I think that many of them will fall out of use
                in the future. JavaScript frameworks like React and Vue make it
                easier to create reusable, style components. Using Sass with
                these frameworks can just create more trouble than it solves.
              </p>
              <p>
                I continue to use and maintain ELR Utilities because
                unfortunately not all of my projects use the most modern,
                cutting edge JavaScript frameworks. Sometimes I just need to
                build a WordPress theme or a landing page quickly. This is where
                a mixin library can really help producivity.
              </p>
            </section>
          </article>
        </main>
      </div>
    );
  }
}

export default App;
