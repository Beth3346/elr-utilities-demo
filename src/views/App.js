import React, { Component } from 'react';
import Header from '../components/Header';
import NavigationMain from '../components/NavigationMain';
import SidebarMain from '../components/SidebarMain';
import ContentHome from '../components/ContentHome';
import './App.css';
// import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavigationMain />
        <Header heading="ELR Utilities">
          A collection of reusable, modular Sass Mixins
        </Header>
        <main className="main-content">
          <SidebarMain />
          <article className="content-holder">
            <ContentHome />
          </article>
        </main>
      </div>
    );
  }
}

export default App;
