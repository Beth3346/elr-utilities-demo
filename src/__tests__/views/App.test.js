import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { createMemoryHistory } from 'history';
// import { withRouter } from 'react-router';
import { Router } from 'react-router-dom';
// this adds custom jest matchers from jest-dom
import 'jest-dom/extend-expect';
import App from '@/App';

afterEach(cleanup);
function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
}

test('full app rendering/navigating', () => {
  const { container } = renderWithRouter(<App />);
  expect(container.innerHTML).toMatch('ELR Utilities');
});
