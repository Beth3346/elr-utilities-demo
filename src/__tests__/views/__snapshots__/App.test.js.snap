// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout has a header 1`] = `"<header class=\\"App-header\\"><div class=\\"header-inner\\"><h1 class=\\"App-title\\">ELR Utilities</h1><p class=\\"App-intro\\">A collection of reusable, modular Scss Mixins</p></div></header>"`;

exports[`layout has main content 1`] = `"<main class=\\"main-content\\"><aside class=\\"App-sidebar\\"><ul class=\\"sidebar-menu\\"><li><a href=\\"/\\">Animations</a><ul><li><a href=\\"/\\">Fade</a></li><li><a href=\\"/\\">Flipper</a></li></ul></li><li><a href=\\"/\\">Blocks</a></li><li><a href=\\"/\\">Components</a></li><li><a href=\\"/\\">CSS Helpers</a></li><li><a href=\\"/\\">Elements</a></li><li><a href=\\"/\\">Flex</a></li><li><a href=\\"/\\">Forms</a></li><li><a href=\\"/grid\\">Grid</a></li><li><a href=\\"/\\">Navigation</a></li><li><a href=\\"/\\">Sass Layout</a></li><li><a href=\\"/\\">Typography</a></li></ul></aside><article class=\\"content-holder\\"><section class=\\"content\\"><h2 class=\\"example-heading\\">Get Started</h2><h3 class=\\"example-subheading\\">Why Create a Sass Mixin Library</h3><p>ELR Utilities is a framework of Sass Mixins and CSS designed to speed up and streamline the development process.</p><p>I created this utility library because I found myself spending a lot of time looking up CSS syntax for stuff I used all the time. Especially clearfix. Using mixins also helps with consistency in CSS styles. If I decide I want to use a different method for styling or learn some new techniques I can update 1 mixin instead of scouring my entire CSS file.</p><h3 class=\\"example-subheading\\">The Future of ELR Utilities and CSS Libraries</h3><p>While I do still think that Sass and CSS Libraries are useful for producivity I think that many of them will fall out of use in the future. JavaScript frameworks like React and Vue make it easier to create reusable, styled components. Using Sass with these frameworks can just create more trouble than it solves.</p><p>I continue to use and maintain ELR Utilities because unfortunately not all of my projects use the most modern, cutting edge JavaScript frameworks. Sometimes I just need to build a WordPress theme or a landing page quickly. This is where a mixin library can really help producivity.</p><h3 class=\\"example-subheading\\">Recent Updates to ELR Utilities</h3><p>Recently I decided to break up ELR Sass Utilities into smaller modules. I noticed that I typically only used a small number of mixins in each project so it really seemed uncessessary to include the entire libaray. All of these modules are available on npm. I have also added compiled CSS to each module for use in projects that don&#x27;t use Scss.</p></section></article></main>"`;

exports[`layout has navigation 1`] = `"<nav class=\\"navigation-main\\"><div class=\\"navigation-inner\\"><p class=\\"logo\\">ELR Utilities</p><ul class=\\"main-menu\\"><li><a href=\\"/\\">CSS</a></li><li><a href=\\"/\\">Sass</a></li><li><a href=\\"/\\">About</a></li><li><a href=\\"/\\">Contact Me</a></li><li><a href=\\"https://github.com/Beth3346\\">GitHub</a></li></ul></div></nav>"`;
